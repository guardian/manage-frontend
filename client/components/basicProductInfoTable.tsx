import { parseDate } from '../../shared/dates';
import type { ProductDetail } from '../../shared/productResponse';
import type { GroupedProductType } from '../../shared/productTypes';
import { ProductDescriptionListTable } from './productDescriptionListTable';

export interface BasicProductInfoTableProps {
	groupedProductType: GroupedProductType;
	productDetail: ProductDetail;
}

export const BasicProductInfoTable = (props: BasicProductInfoTableProps) => {
	return (
		<ProductDescriptionListTable
			content={[
				...(props.groupedProductType.shouldRevealSubscriptionId
					? [
							{
								title: 'Subscription ID',
								value: props.productDetail.subscription
									.subscriptionId,
							},
					  ]
					: []),
				...(props.groupedProductType.tierLabel
					? [
							{
								title: props.groupedProductType.tierLabel,
								value: props.productDetail.tier,
							},
					  ]
					: []),
				...(props.groupedProductType.shouldShowJoinDateNotStartDate
					? [
							{
								title: 'Join date',
								value: parseDate(
									props.productDetail.joinDate,
								).dateStr(),
							},
					  ]
					: [
							{
								title: 'Start date',
								value: props.productDetail.subscription.start
									? parseDate(
											props.productDetail.subscription
												.start,
									  ).dateStr()
									: '-',
							},
					  ]),
			]}
		/>
	);
};
