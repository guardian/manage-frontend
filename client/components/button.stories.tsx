import { brand } from '@guardian/source-foundations';
import type { ComponentMeta, ComponentStory } from '@storybook/react';
import type { ButtonProps } from './buttons';
import { Button } from './buttons';

export default {
	title: 'Components/Button',
	component: Button,
	args: {
		text: 'Button',
		height: undefined,
		fontWeight: undefined,
		left: false,
		right: false,
		disabled: false,
		colour: undefined,
		textColour: undefined,
		primary: undefined,
		hollow: undefined,
		hide: false,
		forceCircle: undefined,
		hoverColour: undefined,
		leftTick: undefined,
		alert: false,
		type: 'button',
	},
	argTypes: {
		fontWeight: {
			options: ['bold', undefined],
			control: { type: 'inline-radio' },
		},
		primary: {
			options: [true, undefined],
			control: { type: 'inline-radio' },
		},
		hollow: {
			options: [true, undefined],
			control: { type: 'inline-radio' },
		},
		forceCircle: {
			options: [true, undefined],
			control: { type: 'inline-radio' },
		},
		leftTick: {
			options: [true, undefined],
			control: { type: 'inline-radio' },
		},
		colour: {
			control: { type: 'color' },
		},
		textColour: {
			control: { type: 'color' },
		},
		hoverColour: {
			control: { type: 'color' },
		},
		type: {
			options: ['button', 'submit', 'reset'],
			control: { type: 'inline-radio' },
		},
	},
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args: ButtonProps) => (
	<Button {...args} />
);

export const Default = Template.bind({});

export const Bold = Template.bind({});
Bold.args = {
	fontWeight: 'bold',
};

export const Disabled = Template.bind({});
Disabled.args = {
	disabled: true,
};

export const WithBrandColours = Template.bind({});
WithBrandColours.args = {
	colour: brand[800],
	textColour: brand[400],
};

export const WithRightArrow = Template.bind({});
WithRightArrow.args = {
	right: true,
};

export const DisabledWithRightArrow = Template.bind({});
DisabledWithRightArrow.args = {
	disabled: true,
	right: true,
};

export const PrimaryWithRightArrow = Template.bind({});
PrimaryWithRightArrow.args = {
	primary: true,
	right: true,
};

export const HollowWithLeftArrow = Template.bind({});
HollowWithLeftArrow.args = {
	hollow: true,
	left: true,
};
