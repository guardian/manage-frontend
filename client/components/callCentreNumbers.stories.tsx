import type { ComponentMeta, ComponentStory } from '@storybook/react';
import type { CallCentreNumbersProps } from './callCentreNumbers';
import { CallCentreNumbers } from './callCentreNumbers';

export default {
	title: 'Components/CallCentreNumbers',
	component: CallCentreNumbers,
	args: {
		prefixText: '',
	},
} as ComponentMeta<typeof CallCentreNumbers>;

const Template: ComponentStory<typeof CallCentreNumbers> = (
	args: CallCentreNumbersProps,
) => {
	return <CallCentreNumbers {...args} />;
};

export const Default = Template.bind({});

export const WithCustomPrefix = Template.bind({});
WithCustomPrefix.args = {
	prefixText:
		'If you’re not sure what’s best for you or would like help, to contact us',
};
