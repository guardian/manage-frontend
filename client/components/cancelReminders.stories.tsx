import { ComponentStory, ComponentMeta } from '@storybook/react';
import fetchMock from 'fetch-mock';
import CancelReminders, { CancelRemindersProps } from './cancelReminders';

export default {
	title: 'Pages/CancelReminders',
	component: CancelReminders,
	parameters: {
		layout: 'fullscreen',
	},
} as ComponentMeta<typeof CancelReminders>;

export const Error: ComponentStory<typeof CancelReminders> = (
	_: CancelRemindersProps,
) => {
	fetchMock.restore().post('/api/reminders/cancel', 500);

	return <CancelReminders reminderCode="123" />;
};

export const Success: ComponentStory<typeof CancelReminders> = (
	_: CancelRemindersProps,
) => {
	fetchMock.restore().post('/api/reminders/cancel', 200);

	return <CancelReminders reminderCode="123" />;
};
