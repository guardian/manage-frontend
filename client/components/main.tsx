import { palette, textSansSizes } from '@guardian/source-foundations';
import type { SignInStatus } from '../services/signInStatus';
import { serif } from '../styles/fonts';
import { Footer } from './footer/footer';
import Header from './header';
import HelpCentreHeader from './helpCentre/helpCentreHeader';

export interface MainProps {
	signInStatus?: SignInStatus;
	requiresSignIn?: boolean;
	helpCentrePage?: boolean;
	children: JSX.Element | JSX.Element[];
}

export const Main = ({
	signInStatus = 'init',
	requiresSignIn,
	helpCentrePage,
	children,
}: MainProps) => (
	<div
		css={{
			display: 'flex',
			flexDirection: 'column',
			height: '100vh',
			alignItems: 'stretch',
			width: '100%',
			color: palette.neutral[20],
		}}
	>
		<a
			css={{
				color: palette.brand[400],
				position: 'absolute',
				textAlign: 'center',
				fontSize: `${textSansSizes.xxsmall}px`,
				':focus': {
					position: 'static',
				},
			}}
			href="#maincontent"
		>
			Skip to main content
		</a>
		{helpCentrePage ? (
			<HelpCentreHeader
				signInStatus={signInStatus}
				requiresSignIn={requiresSignIn}
			/>
		) : (
			<Header
				signInStatus={signInStatus}
				requiresSignIn={requiresSignIn}
			/>
		)}
		<div
			css={{
				flexGrow: 1,
				flexShrink: 0,
				display: 'flex',
				flexDirection: 'column',
			}}
		>
			<main
				css={{
					fontFamily: serif,
					flexGrow: 1,
					flexShrink: 0,
				}}
			>
				{children}
			</main>
		</div>
		<Footer />
	</div>
);
