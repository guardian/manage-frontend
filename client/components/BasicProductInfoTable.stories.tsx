import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { GROUPED_PRODUCT_TYPES } from '../../shared/productTypes';
import { newspaperVoucherPaypal } from '../fixtures/productDetail';
import { BasicProductInfoTable } from './basicProductInfoTable';

export default {
	title: 'Components/BasicProductInfoTable',
	component: BasicProductInfoTable,
} as ComponentMeta<typeof BasicProductInfoTable>;

export const Default: ComponentStory<typeof BasicProductInfoTable> = () => (
	<BasicProductInfoTable
		groupedProductType={GROUPED_PRODUCT_TYPES.subscriptions}
		productDetail={newspaperVoucherPaypal}
	/>
);
