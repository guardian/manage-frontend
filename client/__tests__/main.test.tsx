import { create } from 'react-test-renderer';
import { Main } from '../components/main';

describe('Main', () => {
	beforeEach(() => {
		jest.useFakeTimers();
		jest.setSystemTime(new Date('2022-01-01'));
	});

	afterEach(() => {
		jest.useRealTimers();
	});

	it('renders something', () => {
		const rendered = create(
			<Main>
				<p>hi</p>
			</Main>,
		);

		expect(rendered.toJSON()).toMatchSnapshot();
	});
});
