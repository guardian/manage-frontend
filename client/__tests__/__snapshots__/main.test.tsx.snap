// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders something 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  color: #333333;
}

.emotion-1 {
  color: #052962;
  position: absolute;
  textalign: center;
  fontsize: 12px;
}

.emotion-1:focus {
  position: static;
}

.emotion-2 {
  background-color: #052962;
  min-height: 50px;
  overflow: visible;
  position: relative;
  box-shadow: 0 2px 1px -1px #506991;
}

@media (min-width: 980px) {
  .emotion-2 {
    min-height: 82px;
  }
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  max-width: calc(1300px + 2.5rem);
  margin: auto;
  padding: 0px 12px;
}

@media (min-width: 740px) {
  .emotion-3 {
    padding: 0px 20px;
  }
}

.emotion-4 {
  display: block;
  width: 121px;
  height: 39px;
  text-align: right;
}

@media (min-width: 980px) {
  .emotion-4 {
    width: 158px;
    height: 51px;
  }
}

.emotion-5 {
  width: 121px;
  height: 39px;
}

@media (min-width: 980px) {
  .emotion-5 {
    width: 158px;
    height: 51px;
  }
}

.emotion-6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-7 {
  font-family: GuardianTextEgyptian,"Guardian Text Egyptian Web",Georgia,serif;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-8 {
  background-color: #052962;
  color: #FFFFFF;
}

.emotion-9 {
  max-width: 1300px;
  margin: auto;
}

.emotion-10 {
  padding: 10px;
  border: 1px solid rgba(255, 255, 255, 0.3);
  border-top: 0;
}

@media (min-width: 980px) {
  .emotion-10 {
    padding: 0 20px;
  }
}

@media (min-width: 1140px) {
  .emotion-10 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

.emotion-11 {
  padding: 0;
  border: 0;
  width: 100%;
  margin: 0;
}

@media (min-width: 1140px) {
  .emotion-11 {
    width: 340px;
  }
}

@media (min-width: 1300px) {
  .emotion-11 {
    width: 460px;
  }
}

.emotion-12 {
  min-height: 150px;
}

.emotion-13 {
  font-feature-settings: kern;
  font-size: 16px;
  line-height: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-bottom: 18px;
  border-top: 1px solid rgba(255, 255, 255, 0.3);
}

@media (min-width: 1300px) {
  .emotion-13 {
    border-top: 0;
  }
}

.emotion-14 {
  line-height: 19.2px;
  width: calc(50% - 1.25rem - 1px);
  list-style: none;
  position: relative;
  padding: 0 10px;
  margin: 0;
}

.emotion-14:nth-of-type(even) {
  border-left: 1px solid rgba(255, 255, 255, 0.3);
}

@media (min-width: 740px) {
  .emotion-14 {
    width: 161px;
    -webkit-flex: 1 0 0;
    -ms-flex: 1 0 0;
    flex: 1 0 0;
  }

  .emotion-14:not(:first-of-type) {
    border-left: 1px solid rgba(255, 255, 255, 0.3);
  }
}

@media (min-width: 980px) {
  .emotion-14 {
    border-left: 1px solid rgba(255, 255, 255, 0.3);
  }
}

.emotion-15 {
  list-style: none;
}

.emotion-16 {
  display: inline-block;
  padding: 6px 0;
  color: #FFFFFF;
}

.emotion-16:hover {
  color: #FFE500;
  cursor: pointer;
}

.emotion-57 {
  width: 50%;
  border-left: 1px solid rgba(255, 255, 255, 0.3);
  padding-left: 10px;
  border-top: 1px solid rgba(255, 255, 255, 0.3);
}

@media (min-width: 660px) {
  .emotion-57 {
    width: 312px;
  }
}

@media (min-width: 740px) {
  .emotion-57 {
    border-top: 0;
  }
}

.emotion-58 {
  color: #FFE500;
  font-family: "GH Guardian Headline","Guardian Egyptian Web",Georgia,serif;
  font-size: 1.5rem;
  line-height: 1.15;
  font-weight: 700;
  font-style: normal;
  --source-text-decoration-thickness: 3px;
  line-height: 24px;
  margin-top: 3px;
  margin-bottom: 12px;
}

@media (min-width: 660px) {
  .emotion-58 {
    font-size: 32px;
    line-height: 32px;
  }
}

.emotion-59 {
  display: inline-block;
  margin-right: 10px;
  margin-bottom: 6px;
}

.emotion-60 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  background: transparent;
  cursor: pointer;
  -webkit-transition: .3s ease-in-out;
  transition: .3s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  vertical-align: middle;
  font-family: GuardianTextSans,"Guardian Text Sans Web","Helvetica Neue",Helvetica,Arial,"Lucida Grande",sans-serif;
  font-size: 1.0625rem;
  line-height: 1.3;
  font-weight: 700;
  font-style: normal;
  --source-text-decoration-thickness: 2px;
  height: 36px;
  min-height: 36px;
  padding: 0 16px;
  border-radius: 36px;
  padding-bottom: 2px;
  background-color: #FFE500;
  color: #052962;
}

.emotion-60:disabled {
  cursor: not-allowed;
}

.emotion-60:focus {
  outline: 0;
}

html:not(.src-focus-disabled) .emotion-60:focus {
  outline: 5px solid #0077B6;
  outline-offset: 3px;
}

.emotion-60:hover {
  background-color: #FFD213;
}

.emotion-60 svg {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: block;
  fill: currentColor;
  position: relative;
  width: 26px;
  height: auto;
}

.emotion-60 .src-button-space {
  width: 8px;
}

.emotion-60 svg {
  margin-right: -4px;
}

.emotion-60 svg {
  -webkit-transform: translate(0, 0);
  -moz-transform: translate(0, 0);
  -ms-transform: translate(0, 0);
  transform: translate(0, 0);
  -webkit-transition: .2s cubic-bezier(.64, .57, .67, 1.53);
  transition: .2s cubic-bezier(.64, .57, .67, 1.53);
}

.emotion-60:hover svg,
.emotion-60:focus svg {
  -webkit-transform: translate(2px, 0);
  -moz-transform: translate(2px, 0);
  -ms-transform: translate(2px, 0);
  transform: translate(2px, 0);
}

.emotion-61 {
  padding-bottom: 24px;
  padding-left: 20px;
  padding-right: 20px;
  position: relative;
}

.emotion-62 {
  font-size: 16px;
  color: #FFFFFF;
  font-weight: bold;
  padding: 0 5px;
  background-color: #052962;
  position: absolute;
  right: 20px;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-62:hover {
  color: #FFE500;
}

.emotion-63 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 44px;
  padding-left: 8px;
  font-family: GuardianTextSans,"Guardian Text Sans Web","Helvetica Neue",Helvetica,Arial,"Lucida Grande",sans-serif;
  font-size: 0.9375rem;
  line-height: 1.3;
  font-weight: 700;
  font-style: normal;
  --source-text-decoration-thickness: 2px;
  color: #FFFFFF;
  background-color: #052962;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.emotion-63 :hover {
  color: #FFE500;
}

.emotion-63 :focus {
  outline: 0;
}

html:not(.src-focus-disabled) .emotion-63 :focus {
  box-shadow: 0 0 0 3px #0077B6;
}

.emotion-64 {
  height: 44px;
  width: 44px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 44px;
  background: currentColor;
  margin-left: 8px;
}

.emotion-64 svg {
  height: 26px;
  width: 26px;
  fill: #052962;
}

.emotion-65 {
  padding-top: 26px;
  font-size: 12px;
}

@media (min-width: 740px) {
  .emotion-65 {
    padding-top: 6px;
  }
}

<div
  className="emotion-0"
>
  <a
    className="emotion-1"
    href="#maincontent"
  >
    Skip to main content
  </a>
  <header
    className="emotion-2"
  >
    <div
      className="emotion-3"
    >
      <a
        className="emotion-4"
        href="https://www.theguardian.com"
        title="The Guardian - Back to home"
      >
        <svg
          className="emotion-5"
          fill="none"
          viewBox="0 0 295 95"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M66.2889 50.6378L71.3533 48.0014V8.36507H67.555L58.2402 20.7287H57.2454L57.788 6.91052H98.0317L98.5743 20.7287H97.4891L88.4455 8.36507H84.5568V47.9105L89.5308 50.5469V51.9105H66.2889V50.6378Z"
            fill="#FFFFFF"
          />
          <path
            d="M103.187 48.8182V4.90909L99.2979 3.36364V2.54545L113.315 0H114.762V20.8182L115.124 20.4545C118.199 17.7273 122.721 15.9091 127.152 15.9091C133.302 15.9091 136.015 19.3636 136.015 25.9091V48.8182L139.27 50.6364V52H120.821V50.6364L124.168 48.8182V25.9091C124.168 22.3636 122.63 20.9091 119.646 20.9091C117.656 20.9091 116.028 21.5455 114.762 22.5455V49L118.018 50.8182V52.0909H99.5692V50.8182L103.187 48.8182Z"
            fill="#FFFFFF"
          />
          <path
            d="M151.118 35.3644C151.479 42.6371 154.735 48.2734 162.422 48.2734C166.13 48.2734 168.752 46.5462 171.194 45.2734V46.728C169.295 49.3644 164.412 53.0916 157.629 53.0916C145.782 53.0916 139.632 46.4553 139.632 34.9098C139.632 23.6371 146.234 16.6371 156.996 16.6371C167.125 16.6371 172.37 21.728 172.37 35.0916V35.4553H151.118V35.3644ZM150.937 33.728L161.427 33.0916C161.427 24.0916 159.89 18.0916 156.815 18.0916C153.559 18.0916 150.937 25.0007 150.937 33.728Z"
            fill="#FFFFFF"
          />
          <path
            d="M0 69.4521C0 50.3612 12.5705 43.543 26.588 43.543C32.5567 43.543 38.1637 44.543 41.3289 45.8157L41.6002 59.1793H40.2437L32.0141 46.2702C30.5671 45.6339 29.301 45.4521 26.7689 45.4521C19.3532 45.4521 15.4644 54.0884 15.6453 68.2702C15.8262 85.2702 18.7201 92.9975 25.5932 92.9975C27.4019 92.9975 28.7584 92.7248 29.6628 92.2702V74.0884L25.141 71.4521V69.9975H47.0264V71.6339L42.5046 74.0884V91.9975C38.7967 93.4521 32.5567 94.8157 26.0454 94.8157C10.1288 94.9066 0 87.4521 0 69.4521Z"
            fill="#FFFFFF"
          />
          <path
            d="M46.7549 60.5437V59.4528L61.4958 56.9073L63.1237 56.9982V86.0891C63.1237 89.6346 64.8419 90.6346 67.6454 90.6346C69.4541 90.6346 71.082 89.9073 72.4385 88.3619V62.3619L68.4594 60.6346V59.4528L83.1099 56.8164L84.5568 56.9073V90.271L88.536 91.9073V92.9982L74.0663 94.8164L72.6194 94.7255V90.3619H72.2576C69.5446 92.8164 65.8367 94.9982 61.2245 94.9982C54.1706 94.9982 51.0053 90.8164 51.0053 84.4528V62.3619L46.7549 60.5437Z"
            fill="#FFFFFF"
          />
          <path
            d="M140.627 56.8164L141.803 56.9073V67.6346H142.165C143.702 59.7255 147.229 56.8164 151.389 56.8164C152.022 56.8164 152.836 56.9073 153.198 57.0891V68.18C152.565 67.9982 151.299 67.9073 150.214 67.9073C146.868 67.9073 144.426 68.5437 142.255 69.5437V90.8164L145.601 92.6346V93.9982H126.61V92.6346L130.047 90.7255V61.7255L126.067 60.5437V59.5437L140.627 56.8164Z"
            fill="#FFFFFF"
          />
          <path
            d="M177.435 57.7241V46.3604L173.456 44.9059V43.9968L188.287 41.2695L189.734 41.4513V90.1786L193.804 91.6332V92.9059L179.153 94.9059L177.977 94.815V90.815H177.616C175.445 92.9968 172.551 94.9059 167.939 94.9059C159.981 94.9059 154.103 88.815 154.103 76.2695C154.103 63.0877 160.885 56.5423 171.104 56.5423C174.089 56.5423 176.35 57.0877 177.435 57.7241ZM177.435 88.9059V59.815C176.53 59.1786 175.807 58.4514 173.365 58.5423C169.386 58.7241 166.944 64.7241 166.944 75.4513C166.944 85.0877 168.663 90.4514 173.998 90.2695C175.536 90.1786 176.711 89.6332 177.435 88.9059Z"
            fill="#FFFFFF"
          />
          <path
            d="M209.991 56.815L211.257 56.9059V90.7241L214.603 92.5423V93.9059H195.612V92.5423L199.048 90.6332V62.2695L194.979 60.6332V59.5423L209.991 56.815ZM211.347 47.6332C211.347 51.1786 208.363 53.9059 204.836 53.9059C201.219 53.9059 198.415 51.1786 198.415 47.6332C198.415 44.0877 201.219 41.2695 204.836 41.2695C208.363 41.2695 211.347 44.0877 211.347 47.6332Z"
            fill="#FFFFFF"
          />
          <path
            d="M257.379 90.7266V61.9084L253.4 60.4538V59.0902L268.05 56.3629L269.497 56.4538V60.7266H269.859C273.024 57.9084 277.727 56.0902 282.339 56.0902C288.67 56.0902 291.564 59.0902 291.564 65.8175V90.5448L295 92.4538V93.8175H276.009V92.4538L279.445 90.5448V66.6357C279.445 62.9084 277.817 61.4538 274.833 61.4538C272.934 61.4538 271.306 61.9084 269.769 63.0902V90.8175L273.115 92.7266V94.0902H254.123V92.7266L257.379 90.7266Z"
            fill="#FFFFFF"
          />
          <path
            d="M236.308 72.6346V67.8164C236.308 60.5437 234.77 58.18 230.248 58.18C229.706 58.18 229.254 58.271 228.711 58.271L220.753 69.0891H219.668V59.0891C223.104 57.9982 227.355 56.8164 233.052 56.8164C242.819 56.8164 248.426 59.5437 248.426 67.7255V91.271L251.953 92.18V93.0891C250.596 93.9073 247.793 94.7255 244.809 94.7255C240.016 94.7255 237.664 93.18 236.669 90.4528H236.308C234.228 93.271 231.334 94.8164 226.812 94.8164C221.024 94.8164 217.045 91.18 217.045 84.9073C217.045 78.8164 220.753 75.5437 228.44 74.0891L236.308 72.6346ZM236.308 88.9073V74.3619L233.866 74.5437C230.068 74.9073 228.621 77.3619 228.621 82.7255C228.621 88.6346 230.52 90.18 233.233 90.18C234.77 90.18 235.584 89.7255 236.308 88.9073Z"
            fill="#FFFFFF"
          />
          <path
            d="M108.884 72.6346V67.8164C108.884 60.5437 107.346 58.18 102.825 58.18C102.282 58.18 101.83 58.271 101.287 58.271L93.3289 69.0891H92.2437V59.0891C95.6803 57.9982 99.9307 56.8164 105.628 56.8164C115.395 56.8164 121.002 59.5437 121.002 67.7255V91.271L124.529 92.18V93.0891C123.173 93.9073 120.369 94.7255 117.385 94.7255C112.592 94.7255 110.24 93.18 109.246 90.4528H108.884C106.804 93.271 103.91 94.8164 99.3881 94.8164C93.6002 94.8164 89.6211 91.18 89.6211 84.9073C89.6211 78.8164 93.3289 75.5437 101.016 74.0891L108.884 72.6346ZM108.884 88.9073V74.3619L106.442 74.5437C102.644 74.9073 101.197 77.3619 101.197 82.7255C101.197 88.6346 103.096 90.18 105.809 90.18C107.437 90.18 108.251 89.7255 108.884 88.9073Z"
            fill="#FFFFFF"
          />
        </svg>
      </a>
    </div>
  </header>
  <div
    className="emotion-6"
  >
    <main
      className="emotion-7"
    >
      <p>
        hi
      </p>
    </main>
  </div>
  <footer>
    <div>
      <div
        className="emotion-8"
      >
        <div
          className="emotion-9"
        >
          <div
            className="emotion-10"
          >
            <div
              className="emotion-11"
            >
              <iframe
                className="emotion-12"
                data-form-success-desc="We will send you our picks of the most important headlines tomorrow morning."
                data-node-uid="2"
                frameBorder="0"
                height="86px"
                onLoad={[Function]}
                scrolling="no"
                seamless={false}
                src="https://www.theguardian.com/email/form/footer/today-uk"
                title="Guardian Email Sign-up Form"
              />
            </div>
            <div
              className="emotion-13"
            >
              <ul
                className="emotion-14"
              >
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/about"
                  >
                    About us
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/help/contact-us"
                  >
                    Contact us
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/info/complaints-and-corrections"
                  >
                    Complaints & corrections
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/securedrop"
                  >
                    Secure Drop
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://workforus.thegulocal.com"
                  >
                    Work for us
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="#"
                    onClick={[Function]}
                  >
                    Privacy settings
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/info/privacy"
                  >
                    Privacy policy
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/info/cookies"
                  >
                    Cookie policy
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://www.thegulocal.com/help/terms-of-service"
                  >
                    Terms & conditions
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://www.thegulocal.com/help"
                  >
                    Help
                  </a>
                </li>
              </ul>
              <ul
                className="emotion-14"
              >
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/index/subjects/a"
                  >
                    All topics
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/index/contributors"
                  >
                    All writers
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://uploads.guim.co.uk/2021/07/27/STL_&_GMG_Modern_Slavery_Act_Statement_2021.pdf"
                  >
                    Modern Slavery Act
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://theguardian.newspapers.com/"
                  >
                    Digital newspaper archive
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://www.facebook.com/theguardian"
                  >
                    Facebook
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://twitter.com/guardian"
                  >
                    Twitter
                  </a>
                </li>
              </ul>
              <ul
                className="emotion-14"
              >
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://advertising.thegulocal.com"
                  >
                    Advertise with us
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://thegulocal.com/guardian-labs"
                  >
                    Guardian Labs
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://jobs.thegulocal.com/?INTCMP=NGW_FOOTER_UK_GU_JOBS"
                  >
                    Search jobs
                  </a>
                </li>
                <li
                  className="emotion-15"
                >
                  <a
                    className="emotion-16"
                    href="https://patrons.thegulocal.com/?INTCMP=footer_patrons"
                  >
                    Patrons
                  </a>
                </li>
              </ul>
              <div
                className="emotion-57"
              >
                <div
                  className="emotion-58"
                >
                  Support the Guardian
                </div>
                <div
                  className="emotion-59"
                >
                  <a
                    className="emotion-60"
                    href="https://support.thegulocal.com/contribute?INTCMP=mma_footer_support_contribute&acquisitionData=%7B%22source%22%3A%22GUARDIAN_WEB%22%2C%22componentType%22%3A%22ACQUISITIONS_MANAGE_MY_ACCOUNT%22%2C%22componentId%22%3A%22mma_footer_support_contribute%22%7D&displayExistingPaymentOptions=true"
                    onClick={[Function]}
                  >
                    Support us
                    <div
                      className="src-button-space"
                    />
                    <svg
                      aria-hidden={true}
                      focusable={false}
                      viewBox="-3 -3 30 30"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clipRule="evenodd"
                        d="M1 12.956h18.274l-7.167 8.575.932.932L23 12.478v-.956l-9.96-9.985-.932.932 7.166 8.575H1z"
                        fillRule="evenodd"
                      />
                    </svg>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="emotion-61"
          >
            <div
              className="emotion-62"
            >
              <a
                className="emotion-63"
                href="#top"
              >
                Back to top
                <div
                  className="emotion-64"
                >
                  <svg
                    aria-hidden={true}
                    focusable={false}
                    viewBox="-3 -3 30 30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M23 16.768 12.502 6.27h-1.004L1 16.768l.98.954L12 9.284l10.02 8.438z"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </a>
            </div>
            <div
              className="emotion-65"
            >
              © 
              2022
               Guardian News & Media Limited or its affiliated companies. All rights reserved.
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
